libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.java

```diff
@@ -551,9 +551,9 @@ protected void onPostExecute(OAuth2.TokenEndpointResponse tr) {
                  */
                 if (!passcodeManager.hasStoredPasscode(mgr.getAppContext())) {
                     // This will bring up the create passcode screen - we will create the account in onResume
-                    mgr.getPasscodeManager().setEnabled(true);
-                    mgr.getPasscodeManager().lockIfNeeded((Activity) getContext(), true);
-                } else {
+                    passcodeManager.setEnabled(true);
+                    passcodeManager.lockIfNeeded((Activity) getContext(), true);
+                } else if (!changeRequired) { // If a passcode change is required, the lock screen will have already been set in setMinPasscodeLength
```

Comment:
previously, our code was falling into this 'else' which would add our account and kill the login activity (callback.finish()) before the passcode had been entered, and we again fail to get the activity result.